feat(auth): add new IAM profile selector to VPC Instance Auth (#277)
IBM/go-sdk-core: Authentication.md

### Properties

- IAMProfileCRN: (optional) the crn of the linked trusted IAM profile to be used when obtaining the IAM access token.

- IAMProfileID: (optional) the id of the linked trusted IAM profile to be used when obtaining the IAM access token.

> - IAMProfileName: (optional) the name of the linekd trusted IAM profile to be used when obtaining the IAM access token.

Reviewer: ```suggestion
- IAMProfileName: (optional) the name of the linked trusted IAM profile to be used when obtaining the IAM access token.
```

If not specified by the user, a suitable default Client will be constructed.

Usage Notes:
1. At most one of `IAMProfileCRN`, `IAMProfileID` or `IAMProfileName` may be specified.  The specified value must map
to a trusted IAM profile that has been linked to the compute resource (virtual server instance).

> 2. If more than on from `IAMProfileCRN`, `IAMProfileID` or `IAMProfileName` are specified, then an error occurs.

Reviewer: ```suggestion
2. If more than one of `IAMProfileCRN`, `IAMProfileID` or `IAMProfileName` are specified, then an error occurs.
```
